aopy/postproc.py

```diff
@@ -511,4 +519,146 @@ def get_target_locations(preproc_dir, subject, te_id, date, target_indices):
     for i in range(len(target_indices)):
         trial_idx = np.where(trials['index'] == target_indices[i])[0][0]
         locations.append(trials['target'][trial_idx][[0,2,1]])
-    return np.array(locations)
\ No newline at end of file
+    return np.array(locations)
+
+def estimate_velocity(pos, dt):
+    '''
+    Estimates magnitude of velocity from position data using the central differences derivative method
+
+    Args:
+        pos (nt, nch): array of position data
+        dt (float): sampling period of position data
+        
+    Returns:
+        vel (nt): array of velocity magnitudes
+    '''
+
+    dxyz = np.gradient(pos, axis=0)
+    dist = np.linalg.norm(dxyz, axis=1)
+    vel = dist/dt
+    return vel
+
+def estimate_acceleration(pos, dt):
+    '''
+    Estimates magnitude of acceleration from position data using the central differences derivative method
+    
+    Args:
+        pos (nt, nch): array of position data
+        dt (float): sampling period of position data
+        
+    Returns:
+        acc (nt): array of acceleration magnitudes
+    '''
+    
+    ddxyz = np.gradient(np.gradient(pos, axis=0), axis=0)
+    dist = np.linalg.norm(ddxyz, axis=1)
+    acc = dist/(dt**2)
+    return acc
+
+def get_distance_to_target(pos, target_pos, target_radius):
+    '''
+    Computes distance from position data to boundary of specified target
+    
+    Args:
+        pos (nt, nch): array of position data
+        target_pos (nch): position of center of target
+        target_radius (float): target radius (from metadata)
+        
+    Returns:
+        dist (nt): array of magnitudes of distance from target
+    '''
+    
+    dist_xy = pos-np.tile(target_pos, (pos.shape[0],1))
+    dist = np.linalg.norm(dist_xy, axis=1) - target_radius
+    return dist
+
+def get_average_eye_position(eye_pos):
+    '''
+    Get the average x,y position data between left and right eyes
+    
+    Args:
+        eye_pos (nt, 4): array of eye position data for one trial, as output by exp_data
+    
+    Returns:
+        (nt, 2): array of x,y position data averaged between both eyes
+    '''
+    
+    return np.stack([(eye_pos[:,0]+eye_pos[:,2])/2, (eye_pos[:,1]+eye_pos[:,3])/2], axis=1)
```

Comment:
could we incorporate the eye_labels into the input of this function so it knows exactly what indices to average?